packages/nc-gui/composables/useKanbanViewStore.ts

```diff
@@ -0,0 +1,548 @@
+import type { ComputedRef, Ref } from 'vue'
+import type { Api, ColumnType, KanbanType, SelectOptionType, SelectOptionsType, TableType, ViewType } from 'nocodb-sdk'
+import { useI18n } from 'vue-i18n'
+import { message } from 'ant-design-vue'
+import type { Row } from '~/lib'
+import { SharedViewPasswordInj, deepCompare, enumColor, extractPkFromRow, useInjectionState, useNuxtApp } from '#imports'
+
+type GroupingFieldColOptionsType = SelectOptionType & { collapsed: boolean }
+
+const [useProvideKanbanViewStore, useKanbanViewStore] = useInjectionState(
+  (
+    meta: Ref<TableType | KanbanType | undefined>,
+    viewMeta: Ref<ViewType | KanbanType | undefined> | ComputedRef<(ViewType & { id: string }) | undefined>,
+    shared = false,
+  ) => {
+    if (!meta) {
+      throw new Error('Table meta is not available')
+    }
+
+    const { t } = useI18n()
+
+    const { api } = useApi()
+
+    const { project } = useProject()
+
+    const { $e, $api } = useNuxtApp()
+
+    const { sorts, nestedFilters } = useSmartsheetStoreOrThrow()
+
+    const { sharedView, fetchSharedViewData, fetchSharedViewGroupedData } = useSharedView()
+
+    const { isUIAllowed } = useUIPermission()
+
+    const isPublic = ref(shared) || inject(IsPublicInj, ref(false))
+
+    const password = ref<string | null>(null)
+
+    provide(SharedViewPasswordInj, password)
+
+    // kanban view meta data
+    const kanbanMetaData = ref<KanbanType>({})
+
+    // grouping field column options - e.g. title, fk_column_id, color etc
+    const groupingFieldColOptions = ref<GroupingFieldColOptionsType[]>([])
+
+    // formattedData structure
+    // {
+    //   [val1] : [
+    //     {row: {...}, oldRow: {...}, rowMeta: {...}},
+    //     {row: {...}, oldRow: {...}, rowMeta: {...}},
+    //     ...
+    //   ],
+    //   [val2] : [
+    //     {row: {...}, oldRow: {...}, rowMeta: {...}},
+    //     {row: {...}, oldRow: {...}, rowMeta: {...}},
+    //     ...
+    //   ],
+    // }
+    const formattedData = ref<Record<string, Row[]>>({})
```

Comment:
You can use `Map` here, with `Map` you can have `null` as a key as well. By using map we can avoid keeping it as `uncategorized` and avoid conflict actual option with value `uncategorized`